torch/_inductor/scheduler.py

```diff
@@ -1691,6 +1704,8 @@ def compute_ancestors(self):
         for node in self.nodes:
             ancestors = set()
             for dep in node.unmet_dependencies:
+                if dep.name == node.get_name():
+                    continue
```

Comment:
It seems to be the case that we're just ignoring the read `MemoryDep` everywhere that it comes up. Do the tests pass if we remove it entirely and just rely on the `StarDep`?

Reply:
Where do you suggest we remove the MemoryDep?